pandas/sparse/array.py

```diff
@@ -322,31 +332,51 @@ def take(self, indices, axis=0):
             # return scalar
             return self[indices]
 
-        indices = np.atleast_1d(np.asarray(indices, dtype=int))
-
-        # allow -1 to indicate missing values
+        indices = com._ensure_platform_int(indices)
```

Comment:
Because the impls are different, it is not likely to be merged. One idea is split boundary check to ``common._validate_take_boudary()``?

- https://github.com/pydata/pandas/blob/master/pandas/indexes/base.py#L1369